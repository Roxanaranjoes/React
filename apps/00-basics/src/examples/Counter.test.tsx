import { describe, it, expect } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { Counter } from './Counter';

describe('Counter', () => {
  it('renderiza y snapshot mÃ­nimo', () => {
    const { asFragment } = render(<Counter />);
    expect(screen.getByText('Counter')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it('incrementa y decrementa', () => {
    render(<Counter />);
    const minus = screen.getByRole('button', { name: '-' });
    const plus = screen.getByRole('button', { name: '+' });
    const value = () => screen.getByText(/\d+/);
    expect(value().textContent).toBe('0');
    fireEvent.click(plus);
    expect(value().textContent).toBe('1');
    fireEvent.click(minus);
    expect(value().textContent).toBe('0');
  });
});
