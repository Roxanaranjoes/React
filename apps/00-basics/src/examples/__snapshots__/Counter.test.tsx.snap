// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > renderiza y snapshot m√≠nimo 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="mb-2 text-lg font-semibold"
    >
      Counter
    </h2>
    <div
      class="flex items-center gap-2"
    >
      <button
        class="rounded bg-blue-600 px-3 py-1 text-white"
      >
        -
      </button>
      <span
        aria-live="polite"
        class="w-12 text-center font-mono"
      >
        0
      </span>
      <button
        class="rounded bg-blue-600 px-3 py-1 text-white"
      >
        +
      </button>
    </div>
  </section>
</DocumentFragment>
`;
